torch/csrc/jit/passes/onnx/shape_type_inference.cpp

```diff
@@ -229,25 +229,27 @@ Value* CloneValueFromListConstruct(
   if (elem->cast<IntType>()) {
     scalar_type = at::kLong;
 
-    auto lc_node = v->node();
-    // ListConstruct Int[] output case, we need to transform to ONNX
-    // Concat to ensure the output is a single tensor(dynamic) type in
-    // order to be consumed as inputs
-    std::vector<Value*> unsqueezed;
-    for (auto* input : lc_node->inputs()) {
-      auto new_input = n_graph->addInput();
-      new_input->copyMetadata(input);
-      Node* unsqueezed_node = createONNXUnsqueeze(
-          n_graph.get(), n_graph->return_node(), new_input, 0, opset_version);
-      unsqueezed.emplace_back(unsqueezed_node->output());
-    }
-    Node* concat_node =
-        n_graph->insertNode(n_graph->create(::c10::onnx::Concat, 1));
-    concat_node->i_(attr::axis, 0);
-    for (auto v : unsqueezed) {
-      concat_node->addInput(v);
-    }
-    return concat_node->output();
+    if (!v->node()->inputs().empty()) {
+      auto lc_node = v->node();
```

Comment:
I don't think the discussion here is resolved. Is it possible to abstract out the logic in both peephole and shape inference passes?

Reply:
Makes sense, and we can do this.